src/app/model/ChessEngine.ts

```diff
@@ -37,8 +38,9 @@ export class ChessEngine implements ChessModel {
     };
     const pieceWasCaptured = this.pieceWasCaptured(squareTo, chosenPiece);
 
+    const pawnPromotionWasEnabled = chosenPiece instanceof Pawn ? this.pawnPromotionWasEnabled(chosenPiece, squareTo) : undefined;
```

Comment:
To sprawdzenie czy to Pionek to moze warto wszystko zawrzec w metodzie this.pawnPromotionWasEnabled(chosenPiece, squareTo) ? Jak myslisz?

Reply:
Myślałam o tym, ale nie wiem czy jest sens w ogóle "wchodzić" w tę metodę, jak bierka nie jest pionem.

Reply:
Jak sadzisz :) Chociaz zdaje mi sie OK, jak mamy to zachowanie dotyczace promocji wyciagniete cale do metody. 
Podobnie jak `this.pieceWasCaptured` - fajnie byloby zachowac jedno rozwiazanie miedzy tymi metodami, bo problem jest ten sam.

Reply:
Spoko, to zmienię, choć widzę różnicę w **piece**WasCaptured a **pawn**PromotionWasEnabled ;)

Reply:
A tak - tak. Możesz mi chodzi zmienić nazwę a zostawić duplikacje kodu. 
Ważniejsze jest - zeby nie mieć duplikacji znaczenia :) Ten sam kod w 2 miejscach nie zawsze jest zły. Ważniejsze jest, żeby nazwy były odpowiednie i adekwatne do tego co robią. A event hasOccurred / isDefined trochę sie różni :) 

pawnPromotionWasEnabled - aaa no rzeczywiście. To jak sadzisz :) Jednakże, skoro można z metody zwrócić undefined, to jak dla mnie jest spoko ubrać to w metodę.

Reply:
done